regression_detector.py

```diff
@@ -158,6 +159,31 @@ def process_regressions_into_gh_issue(regressions_dict, owner: str, output_path:
         f.write(issue_body)
 
 
+def get_best_start_date(latest_metrics_jsons, end_date: str) -> Optional[datetime]:
+    """Get the date closest to `end_date` from `latest_metrics_jsons`"""
+    end_datetime = datetime.strptime("%Y-%m-%d", end_date)
+    for metrics_json in latest_metrics_jsons:
+        start_datetime = datetime.strptime(metrics_json.split('/')[-1].split('-')[-1].split('.')[0], '%Y%m%d%H%M%S')
```

Comment:
We should probably abstract out a function for this by now haha, maybe something called get_date_from_metrics_filename

```
metrics_json.split('/')[-1].split('-')[-1].split('.')[0]
```

Reply:
Good point!